codeChallenges/challenge-4/index.test.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Complete the method/function so that it converts dash/underscore delimited words or any type from camel casing to snake case
+ * The first word within the output should be capitalized only if the original word was capitalized (known as Upper Camel Case, also often referred to as Pascal case).
+ */
+
+/**
+ * Examples
+  "theStealthWarrior"  gets converted to "the-stealth-warrior"
+  "TheStealthWarrior"  gets converted to "The_Stealth_Warrior"
+ */
+
+  function toSnakeCase(str) {
+    if (str.length == 0){
+        return str;
+    }
+    else{
+        let toLower = str.toLowerCase()
```

Comment:
here you transform the text to lower case 
and down you are splitting it based on finding upper case charecter .?